pylint/checkers/typecheck.py

```diff
@@ -1629,6 +1631,45 @@ def visit_unaryop(self, node):
             # Let the error customize its output.
             self.add_message("invalid-unary-operand-type", args=str(error), node=node)
 
+    @check_messages("unsupported-binary-operation")
+    def visit_binop(self, node: astroid.BinOp):
+        # Test alternative Union syntax PEP 604 - int | None
+        msg = "unsupported operand type(s) for |"
+        if node.op == "|" and (
+            not is_postponed_evaluation_enabled(node)
+            and isinstance(
+                node.parent, (astroid.AnnAssign, astroid.Arguments, astroid.FunctionDef)
+            )
+            or not PY310_PLUS
+            and isinstance(
+                node.parent,
+                (
+                    astroid.Assign,
+                    astroid.Call,
+                    astroid.Keyword,
+                    astroid.Dict,
+                    astroid.Tuple,
+                    astroid.Set,
+                    astroid.List,
+                ),
+            )
+        ):
+            for n in (node.left, node.right):
+                n = helpers.object_type(n)
+                if isinstance(n, astroid.ClassDef) and (
+                    n.name == "type"
+                    or any(
+                        [
```

Comment:
We can also remove the list comprehension for better performance (so we do not create the list and just cut short at the first True) :)

Reply:
Changed it